In [5]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score


In [11]:
df = pd.read_csv('house price data.csv')
df.head()





TypeError: 'tuple' object is not callable

In [13]:
df.shape

(4600, 18)

In [15]:
data_cleaned = df.dropna().drop_duplicates()
data_cleaned.shape

(4600, 18)

In [16]:
# Select relevant features and target
features = ['bedrooms', 'bathrooms', 'sqft_living', 'sqft_lot', 'floors', 
            'waterfront', 'view', 'condition', 'sqft_above', 'sqft_basement', 
            'yr_built', 'yr_renovated']
target = 'price'

X = data_cleaned[features]
y = data_cleaned[target]

In [19]:
#Spliting the data
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

In [18]:
model = LinearRegression()
model.fit(X_train, y_train)


LinearRegression()

In [21]:
# Check the accuracy of the model on the training dataset
train_predictions = model.predict(X_train)
train_mse = mean_squared_error(y_train, train_predictions)
train_r2 = r2_score(y_train, train_predictions)

print(f'Training Mean Squared Error: {train_mse}')
print(f'Training R-squared: {train_r2}')



Training Mean Squared Error: 64764831210.21875
Training R-squared: 0.544185359314763


In [23]:
# Test the model using the test dataset
test_predictions = model.predict(X_test)
test_mse = mean_squared_error(y_test, test_predictions)
test_r2 = r2_score(y_test, test_predictions)

print(f'Test Mean Squared Error: {test_mse}')
print(f'Test R-squared: {test_r2}')

Test Mean Squared Error: 986921767056.1471
Test R-squared: 0.03228385663275535


# The results from applying the linear regression model are as follows:

Training Dataset
-- Mean Squared Error (MSE): 64,764,831,210.22
-- R-squared (R²): 0.5442
Testing Dataset
-- Mean Squared Error (MSE): 986,921,767,056.11
-- R-squared (R²): 0.0323



# Conculsion 
The model shows a moderate fit on the training data with an R² value of 0.5442, indicating that approximately 54.42% of the variability in the training data is explained by the model. However, the performance on the test data is poor, with a very low R² value of 0.0323, suggesting that the model does not generalize well to unseen data.

This indicates that the model may be overfitting the training data and not capturing the underlying patterns accurately. Further steps such as feature engineering, regularization, or using more complex models might be necessary to improve the performance. ​